File: aliandinnocents_hackerearth.java
import java.util.*;
public class aliandinnocents_hackerearth{
    public static void main(String[] args){
        Scanner s = new Scanner(System.in);         
        String name = s.nextLine();             //DDXDDD-DD
        char vowel = name.charAt(2);
        int a = (int)name.charAt(0) + (int)name.charAt(1);
        int b = (int)name.charAt(3) + (int)name.charAt(4);
        int c = (int)name.charAt(4) + (int)name.charAt(5);
        int d = (int)name.charAt(7) + (int)name.charAt(8);
        if(vowel == 'A' || vowel == 'E' || vowel == 'I' || vowel == 'O' || vowel == 'U' || vowel == 'Y'){
            System.out.println("invalid");
        }
        else if(a % 2 != 0 || b % 2 != 0 || c % 2 != 0 || d % 2 != 0){
            System.out.println("invalid");
        }
        else{
            System.out.println("valid");
        }
    }
}

File: aliandinnocents_hackerearth.py
name = input()
#DDXDDD-DD
if(name[2]=='A' or name[2]=='E' or name[2]=='I' or name[2]=='O' or name[2]=='U' or name[2]=='Y'):
    print("invalid")
else:
    check = int(name[0])+int(name[1])
    if(check%2==0):
        check = int(name[3])+int(name[4])
        if(check%2==0):
            check = int(name[4])+int(name[5])
            if(check%2==0):
                check = int(name[7])+int(name[8])
                if(check%2==0):
                    print("valid")
                else:
                    print("invalid")
            else:
                print("invalid")
        else:
            print("invalid")
    else:
        print("invalid")